src/deltacode/__init__.py

```diff
@@ -67,6 +85,48 @@ def __init__(self, new_path, old_path, options):
             self.deltas.sort(key=lambda Delta: Delta.factors, reverse=False)
             self.deltas.sort(key=lambda Delta: Delta.score, reverse=True)
 
+    def get_files(self,codebase,is_new):
+        """
+        Walk through the codebase, then generate the resources it(including all files and its directories)
+        then we enumerate over this generated codebase to get file, and directories as (obj)
+        Now during the time of enumeration we append files in the self.new_files list and incremants out self.new_files_count
+        Similarly for old_files.
+        Now , we also maintain a map which maps from object path to its fingerprint.
+        This map will be required when we calculate the hamming distances and compare similarity.
+        """
+        resources = codebase.walk_filtered(topdown=True)
+        for i,obj in enumerate(resources):            
+            if is_new:
+                # append in the new_files
+                self.new_files.append([obj,''])
+                try :
+                    self.new_files_fingerprint[obj.path] = obj.fingerprint
+                except AttributeError:
+                    self.new_files_fingerprint[obj.path] = None
+                if obj.is_file:
+                    # increment the new files count
+                    self.new_files_count += 1
```

Comment:
Why not use `dict.get()` method instead of catching exception?

Reply:
@majurg it would be nice  , changing it

Reply:
@majurg , the `dict.get()` method can not be used here as `obj` is not a `dictionary` it is a `Virtualcodebase` object.